### PR TITLE
Alerting: Update fixed roles to include silences permissions

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What is this feature?**
This PR updates the fixed roles `fixed:alerting.rules:reader` and `fixed:alerting.rules:writer` to include silence permissions introduced in https://github.com/grafana/grafana/pull/85418. 
Also, it updates managed roles to include silence permissions in the folder scope along with rule's permissions.

Currently, no API uses this set of permissions. The API will be updated in the future PRs.

**Why do we need this feature?**
This is needed to support rule-specific silences.

**Who is this feature for?**
Alerting users.

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by IevaVasiljeva — COMMENTED** [review]
How do we currently control access to silences? Is it just by checking `alert.instances:create` etc permissions, or do we also check access to folders or anything else?

My worry is that there might be some custom permission setups that need to be considered when migrating the silence permissions. For instance, permission restrictions (what if Viewers don't have access to read silences, and Editors to create and update them?) and permission widening (what if Viewers have been granted access to create silences?). Seems like the migration is only taking into account user's access to folders, but not user's current access to silences.

And then there's the usual problem with upgrades and downgrades - if the instance is downgraded and then upgraded again after a while, the users will lose access to silences for any folders that have been created in between. But we don't have a good solution to it (apart from running the migration on every startup), so perhaps documenting it is the best thing that we can do.

**Comment by yuri-tceretian** [comment]
The silence permission is not used anywhere as of now, and it's part of a new feature that will let users with access to particular rules in a folder create silences for those rules without affecting rules they do not have access. The current way of doing it is to give `alert.instances:create` which grants unscoped access to silences, so users may create silences that accidentally silence rules they do not have access to. These permissions are not affected and will not go anywhere in the future. 

The new silence permissions have folder scope and are satellite to rule's permissions, which are managed via folder permissions. That's why migration is for. We decided to not use rule's permission in silence API to have more scoped permissions and more flexibility in custom roles definitions.

The custom roles are not taken care of because usually they are provisioned and migration will not work for them. We will add this note to the changelog when the feature is released.

**Review by jdbaldry — APPROVED** [review]
docs lgtm

**Review by IevaVasiljeva — COMMENTED** [review]
> The current way of doing it is to give alert.instances:create which grants unscoped access to silences, so users may create silences that accidentally silence rules they do not have access to. These permissions are not affected and will not go anywhere in the future.

So if someone has removed `alert.instances:create` from a basic role (eg, from the Editor role), would a user with the `Editor` role now be able to create alert silences in folders that they can edit?

If not, then all looks good to me! And good shout on adding a release note about custom roles. Probably also worth mentioning that users will need to rerun this migration if they downgrade to an older version and create a folder or set folder permissions, and then upgrade again.

**Comment by yuri-tceretian** [comment]
> So if someone has removed alert.instances:create from a basic role (eg, from the Editor role), would a user with the Editor role now be able to create alert silences in folders that they can edit?

That is what we count on. The reason why someone would remove `alert.instances:create|write` from users is to prevent them from creating silences that could affect alert rules the user may not see or may not understand that they will be affected (for example, in the case when they are not alerting at the time the silence was created). The permission we grant in this PR will let them create a silence that 100% affects only a single rule.
To summarize, permission `alert.silences` (in folder scope) allows users to create silences that affect only a single rule whereas permission `alert.instances` grants the user the ability to create general silences that may affect many rules. We want the former permission to be granted for all rules the user has access to.

**Comment by IevaVasiljeva** [comment]
> > So if someone has removed alert.instances:create from a basic role (eg, from the Editor role), would a user with the Editor role now be able to create alert silences in folders that they can edit?
> 
> That is what we count on. The reason why someone would remove `alert.instances:create|write` from users is to prevent them from creating silences that could affect alert rules the user may not see or may not understand that they will be affected (for example, in the case when they are not alerting at the time the silence was created). The permission we grant in this PR will let them create a silence that 100% affects only a single rule. To summarize, permission `alert.silences` (in folder scope) allows users to create silences that affect only a single rule whereas permission `alert.instances` grants the user the ability to create general silences that may affect many rules. We want the former permission to be granted for all rules the user has access to.

Thanks for the explanation! I think it might be worth adding a disclaimer (maybe a breaking change notice?) just in case there are cases where the user should not be able to interact with any silences at all (even the ones linked to the folders that they can CRUD).

**Review by IevaVasiljeva — APPROVED** [review]
Approving, as the code looks good. I've added some comments about how this could potentially grant access to silences for users who shouldn't have it. But I think it's fair to assume that access has mostly been revoked because it applied globally, so I will trust you with that judgment :)
